src/services/codefixes/fixMissingTypeAnnotationOnExports.ts

```diff
@@ -0,0 +1,1192 @@
+                element,
+                parent,
+                expression: { kind: ExpressionType.ARRAY_ACCESS, arrayIndex: i },
+            });
+        }
+    }
+
+    function addObjectBindingPatterns(bindingPattern: ObjectBindingPattern, bindingElements: ExpressionReverseChain[], parent: ExpressionReverseChain) {
+        for (const bindingElement of bindingPattern.elements) {
+            let name: string;
+            if (bindingElement.propertyName) {
+                if (isComputedPropertyName(bindingElement.propertyName)) {
+                    bindingElements.push({
+                        element: bindingElement,
+                        parent,
+                        expression: { kind: ExpressionType.COMPUTED, computed: bindingElement.propertyName.expression },
+                    });
+                    continue;
+                }
+                else {
+                    name = bindingElement.propertyName.text;
+                }
+            }
+            else {
+                name = (bindingElement.name as Identifier).text;
+            }
+            bindingElements.push({
+                element: bindingElement,
+                parent,
+                expression: { kind: ExpressionType.TEXT, text: name },
+            });
+        }
+    }
+
+    function createChainedExpression(expression: ExpressionReverseChain, expressionToVar: Map<Expression, Identifier>): Expression {
+        const reverseTraverse: ExpressionReverseChain[] = [expression];
+        while (expression.parent) {
+            expression = expression.parent;
+            reverseTraverse.push(expression);
+        }
+        let chainedExpression: Expression = (reverseTraverse[reverseTraverse.length - 1].expression as { identifier: Identifier; }).identifier;
+        for (let i = reverseTraverse.length - 2; i >= 0; --i) {
+            const nextSubExpr = reverseTraverse[i].expression;
+            if (nextSubExpr.kind === ExpressionType.TEXT) {
+                chainedExpression = factory.createPropertyAccessChain(
+                    chainedExpression,
+                    /*questionDotToken*/ undefined,
+                    factory.createIdentifier(nextSubExpr.text),
+                );
+            }
+            else if (nextSubExpr.kind === ExpressionType.COMPUTED) {
+                chainedExpression = factory.createElementAccessExpression(
+                    chainedExpression,
+                    expressionToVar.get(nextSubExpr.computed)!,
+                );
+            }
+            else if (nextSubExpr.kind === ExpressionType.ARRAY_ACCESS) {
+                chainedExpression = factory.createElementAccessExpression(
+                    chainedExpression,
+                    nextSubExpr.arrayIndex,
+                );
+            }
+        }
+        return chainedExpression;
+    }
+    interface InferenceResult {
+        typeNode?: TypeNode | undefined;
+        mutatedTarget: boolean;

```

Comment:
This looks spooky to me 😅

Reply:
I'm assuming you're talking about the naming or would it be the logic?

Reply:
I mean the strategy of having functions that conditionally mutate something and have to communicate whether they've done so... I didn't study it enough to fully understand why that was needed. Not necessarily a blocker, just a little eyebrow-raising